src/Diagnostics.DataProviders/DataProviders/ChangeAnalysisDataProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading.Tasks;
+using Diagnostics.DataProviders.DataProviderConfigurations;
+using Diagnostics.ModelsAndUtils.Models.ChangeAnalysis;
+using Diagnostics.DataProviders.Interfaces;
+
+namespace Diagnostics.DataProviders
+{
+    public class ChangeAnalysisDataProvider : DiagnosticDataProvider, IChangeAnalysisDataProvider
+    {
+        private ChangeAnalysisDataProviderConfiguration dataProviderConfiguration;
+
+        private ChangeAnalysisClient changeAnalysisClient;
+
+        private string dataProviderRequestId;
+
+        private KustoDataProvider kustoDataProvider;
+
+        public ChangeAnalysisDataProvider(OperationDataCache cache, ChangeAnalysisDataProviderConfiguration configuration, KustoDataProviderConfiguration kustoConfig, string requestId, string clientObjectId, string principalName) : base(cache)
+        {
+            dataProviderConfiguration = configuration;
+            dataProviderRequestId = requestId;
+            changeAnalysisClient = new ChangeAnalysisClient(clientObjectId, principalName);
+            kustoDataProvider = new KustoDataProvider(cache, kustoConfig, requestId);
+        }
+
+        /// <summary>
+        /// Get all change sets for the given arm resource uri.
+        /// </summary>
+        /// <param name="armResourceUri">ARM Resource URI.</param>
+        /// <param name="startTime">Start time of time range.</param>
+        /// <param name="endTime">End time of the time range.</param>
+        /// <returns>List of changesets.</returns>
+        public async Task<List<ChangeSetResponseModel>> GetChangeSetsForResource(string armResourceUri, DateTime startTime, DateTime endTime)
+        {
+            ChangeSetsRequest request = new ChangeSetsRequest
+            {
+                ResourceId = armResourceUri,
+                StartTime = startTime,
+                EndTime = endTime
+            };
+
+            // Get changeSet of the given arm resource uri
+            return await changeAnalysisClient.GetChangeSetsAsync(request);
+        }
+
+        /// <summary>
+        /// Get all Changes for a given changesetId.
+        /// </summary>
+        /// <param name="changeSetId">ChangeSetId retrieved from <see cref="GetChangeSetsForResource(string, DateTime, DateTime)"/>.</param>
+        /// <param name="resourceUri">ARM Resource Uri.</param>
+        /// <returns>List of changes.</returns>
+        public async Task<List<ResourceChangesResponseModel>> GetChangesByChangeSetId(string changeSetId, string resourceUri)
```

Comment:
Can you have overloaded method that just takes `ChangeSetResponseModel`?

Reply:
The idea is like this:
1. We get all change sets and display in UI in a way that would prompt user to click on each one of them
2. When the user clicks on a particular changeset, we fetch changes related to it. More like on-demand fetching changes. We have to keep in mind that each "changeset" can return 9-10 changes. 

Reasons behind this approach is:
Performance impact: If we query all changesets at once, it will cause throttling because ChangeAnalysis API needs to make 3-4 API calls at their end to fetch changes. The experience will be slow. 

So the idea is to get changesets separately and then fetch changes.